packages/headless/src/api/analytics/search-analytics.ts

```diff
@@ -285,22 +286,28 @@ export class SearchAnalyticsProvider
   }
 }
 
-interface LegacyConfigureAnalyticsOptions {
+interface LegacyConfigureAnalyticsOptions<
+  State extends ConfigurationSection = StateNeededBySearchAnalyticsProvider,
```

Comment:
I'm curious and don't 100% understand the ramifications of changing this, why are we allowing State to be `ConfigurationSection` when we already know the type that this supports essentially?

```
export type StateNeededBySearchAnalyticsProvider = ConfigurationSection &
  Partial<Omit<SearchAppState, 'configuration'>>;
```

Reply:
I'll pretend I understand deeply enough TypeScript composes types and generics and agree with the fact that it works. :)